dnf5-plugins/reposync_plugin/reposync.hpp

```diff
@@ -0,0 +1,73 @@
+/*
+Copyright Contributors to the libdnf project.
+
+This file is part of libdnf: https://github.com/rpm-software-management/libdnf/
+
+Libdnf is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+Libdnf is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+
+#ifndef DNF5_PLUGINS_REPOSYNC_PLUGIN_REPOSYNC_HPP
+#define DNF5_PLUGINS_REPOSYNC_PLUGIN_REPOSYNC_HPP
+
+
+#include <dnf5/context.hpp>
+#include <libdnf5-cli/session.hpp>
+#include <libdnf5/repo/repo.hpp>
+#include <libdnf5/rpm/package.hpp>
+
+#include <filesystem>
+#include <string>
+#include <unordered_set>
+#include <vector>
+
+namespace dnf5 {
+
+
+class ReposyncCommand : public Command {
+public:
+    explicit ReposyncCommand(Context & context) : Command(context, "reposync") {}
+    void set_parent_command() override;
+    void set_argument_parser() override;
+    void configure() override;
+    void run() override;
+
+private:
+    using download_list_type = std::vector<std::pair<std::filesystem::path, libdnf5::rpm::Package>>;
```

Comment:
I may be missing something but would `std::map<std::filesystem::path, libdnf5::rpm::Package>` be a better fit here?

The paths from the vector are converted to an `unordered_set` twice in the code, using the map we could avoid that.

Though the order would be different.

Reply:
Thank you! The use of vector here is most likely the result of gradual development. Let me check whether the `std::map` would be a better fit.